e2e/shortcut-action/shortcut-action.e2e-spec.ts

```diff
@@ -0,0 +1,86 @@
+        expect(newTableRowsNum).toEqual(tableRowsNum + 1);
+      });
+  });
+
+  it('should move row up using `mod+shift+up` shortcut', () => {
+    let inputElem = page.getInputElementById('accelerator_experiments.1.experiment');
+    let inputElemValue = inputElem.getAttribute('value');
+    hotkeys.trigger(`${mod}+shift+up`, { targetElement: inputElem });
+    let movedUpInputElem = page.getInputElementById('accelerator_experiments.0.experiment');
+    expect(movedUpInputElem.getAttribute('value')).toEqual(inputElemValue);
+  });
+
+  it('should move row down using `mod+shift+down` shortcut', () => {
+    let inputElem = page.getInputElementById('accelerator_experiments.0.experiment');
+    let inputElemValue = inputElem.getAttribute('value');
+    hotkeys.trigger(`${mod}+shift+down`, { targetElement: inputElem });
+    let movedDownInputElem = page.getInputElementById('accelerator_experiments.1.experiment');
+    expect(movedDownInputElem.getAttribute('value')).toEqual(inputElemValue);
+  });
+
+  it('should delete the current row in table  using `ctrl+backspace` shortcut', () => {
+    let inputElem = page.getInputElementById('keywords.0.keyword');
+    let inputElemValue = inputElem.getAttribute('value');
+    hotkeys.trigger(`${mod}+backspace`, { targetElement: inputElem });
+    let inputElemAfterDeletion = page.getInputElementById('keywords.0.keyword');
+    expect(inputElemAfterDeletion.getAttribute('value')).not.toEqual(inputElemValue);
+  });
+
+  it('should copy new row under keywords.0 table using `alt+c shortcut. ' +
+    'The copied cell under the focused one must be empty and the remaining cells must be exactly copied.', () => {
+    let firstColInputElemValue = page.getInputElementById('keywords.0.classification_scheme').getAttribute('value');
+    let secondColInputElemValue = page.getInputElementById('keywords.0.keyword').getAttribute('value');
+    hotkeys.trigger('alt+c', { targetElement: firstColInputElemValue });
+    let firstColAfterCopyInputElemValue = page.getInputElementById('keywords.1.classification_scheme').getAttribute('value');
+    let secondColAfterCopyInputElemValue = page.getInputElementById('keywords.1.keyword').getAttribute('value');
+    expect(firstColAfterCopyInputElemValue).toEqual('');
+    expect(secondColAfterCopyInputElemValue).toEqual(secondColInputElemValue);
+  });
+
+  it('should copy new row under references table using `ctrl+shift+c` shortcut. ' +
+    'It must copy the exact value of the root element eg Copy the whole author under the focused one.', () => {
+    let inputElemValue = page.getInputElementById('authors.0.affiliations.0.value').getAttribute('value');
+    hotkeys.trigger(`${mod}+shift+c`, { targetElement: inputElemValue });
+    let afterCopyInputElemValue = page.getInputElementById('authors.1.affiliations.0.value').getAttribute('value');
+    expect(inputElemValue).toEqual(afterCopyInputElemValue);
```

Comment:
You can also count that the number of `input` and `textarea` equals the element copied.